File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
from collections import defaultdict

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        res = 0
        for n in nums:
            if n in counts and counts[n]:
                continue
            counts[n] = counts.get(n-1,0) + counts.get(n+1,0) + 1
            counts[n - counts[n-1]] = counts[n]
            counts[n + counts[n+1]] = counts[n]
            res = max(res, counts[n])

        return res

File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int s = 0;
        int res=INT_MIN;
        for(auto& n: nums){
            s = max(n, n+s);
            res = max(res, s);
        }
        return res;
    }
};
